iam.tf

```diff
@@ -0,0 +1,26 @@
+data "azurerm_user_assigned_identity" "uai" {
+  count               = var.aks_identity == "uai" ? ( var.aks_uai_name == null ? 0 : 1 ) : 0
+  name                = var.aks_uai_name
+  resource_group_name = local.network_rg.name
+}
+
+resource "azurerm_user_assigned_identity" "uai" {
+  count               = var.aks_identity == "uai" ? ( var.aks_uai_name == null ? 1 : 0 ) : 0
```

Comment:
Should the last 0 (zero) actually be a 1 with the this logic? I am reading,

if aks_identity == uai then
  if aks_uai_name is null then
    count = 1
  else
    count = 0
else 
  count = 0 <-- I believe this should be 1 given the identity is set so it should be created.

Reply:
the new variable aks_identity=sp|uai tells us whether the user wants to use the Service Principal or an Identity. So, if aks_identity is NOT UAI, then we don't want create an UAI here.

Reply:
Got it.